### ISSUE TITLE
ME2 not applying menu.msgbnd.dcx

### ISSUE THREAD

**Issue by ShaneD20** [Nordgaren/ERBingoRandomizer#6]
I made a fork to maintain a version of the randomizer for a few Elden Ring bingo groups after the initial DLC updates.

- After the DLC update the class descriptions are still not being modified in game (menu.msgbnd.dcx)
- As I look through the code it should be modified with all the randomized gear, as ME2 has no problem using regulation.bin.
- The relative path for the file is still bingo/msg/engus

**Comment by Nordgaren** [comment]
This is clearly an ME2 problem...

**Comment by ShaneD20** [comment]
Yeah, that's my concern. I was hoping you might have some insight. All good though.

**Comment by Nordgaren** [comment]
Actually, I might. I take back what I said :fatcat:

But IDK how soon I can get to it. I am working on the next season and I am incredibly ill right now. If I find a way to fix it for next season, I will merge the changes in, and let you know.

**Comment by ShaneD20** [comment]
I hope you recover from your illness soon. 
(Whether you focus on this or another project after is up to you, again I hope you have a swift and comfortable recovery)

**Comment by Nordgaren** [comment]
It's because the game uses the dlc bnds, now, and not the originals. I'll post the update today, but I can't test the implementation I have on this particular repo, but it should be fine.

Also to note, there's really not any accounting for DLC at all in this, so, none of that is gonna be touched. I have been doing pretty awful, lately, unfortunately. Sick every morning. Trying to get normal work done to stay afloat, and my main mod, ErdTools, is broken.

**Comment by ShaneD20** [comment]
In early testing updating MenuMsgBNDPath has fixed the issue.
 
I haven't done any testing with updating ItemMsgBNDPath yet, but I'm going to investigate if it might affect another bug I was working on (potentially).

Many thank yous Nord, I really appreciate this!

I'm sorry to hear you're still going through this. I hope you can get relief soon and I hope things go well with ErdTools.

**Comment by Nordgaren** [comment]
Had you updated SoulsFormats for usage with the new param format? I was trying to add Ivis SoulsFormatsNEXT, but, I can't use the SFUtil in that, so, I might need to make my own fork, or ask Ivi if they can make it public. I don't know if I can inherit from a static class, or inherit static functions from a class. haven't tried it, yet.

**Comment by Nordgaren** [comment]
Nope. can't inherit from ParamUtil (which is the one I actually need)

**Comment by ShaneD20** [comment]
I'm still new to souls modding, so I only got enough updated in SoulsFormats for the new compression type. I've only recently got somewhat comfortable with the params in base-game. If I come across a solution I will let you know.

**Comment by Nordgaren** [comment]
My suggestion would be to use this fork of [SoulsFormats](https://github.com/soulsmods/SoulsFormatsNEXT/). You will just need to make the ParamUtil public and maybe another class.. I'll post an update later. I have to add it as a submodule.

**Comment by ShaneD20** [comment]
I'll take your advice on this. I'll try to implement it when I have the time to invest in updating to SoulsFormatsNext.

**Comment by ShaneD20** [comment]
I've got good news, SoulsFormatsNext is working with the project.
- I did have to make ParamUtil public, as well as one other class
- Lastly, I had to implement --> `public void WriteBytes(Span<byte> values) {}` in BinaryWriterEx.cs

The first implementation I tried seems to be working. I haven't had the time to test out more things, but the randomizer is working with SoulsFormatsNext.

**Comment by Nordgaren** [comment]
```
public void WriteBytes(Span<byte> values) {} in BinaryWriterEx.cs
```

Where was this for?

**Comment by ShaneD20** [comment]
In FSParam/Param.cs `protected override void Write(BinaryWriterEx bw) {}`
Rows 514 and 515:
`var data = _paramData.DataForElement(Rows[i].DataIndex); 
bw.WriteBytes(data);`

data is a `Span<byte>`

**Comment by Nordgaren** [comment]
That is on line 887 for me. Are you sure you have the right FSParam?

**Comment by ShaneD20** [comment]
Apologies, I forgot that I had moved Cell, Column, and Row to separate files.
![FSParam](https://github.com/user-attachments/assets/b9f4e65a-fde5-4562-a896-38f31d20a0ab)

So the line number should be different.

**Comment by Nordgaren** [comment]
hmm, I didn't have to do this, at all, for my implementation. Did you change anything else that might have effected that?

**Comment by ShaneD20** [comment]
Potentially, there were quite a few refactors to make sure I understood the code base. As well as me adding code to reduce smithing level, etc.

**Comment by ShaneD20** [comment]
I've found another odd bug. Let me know if you'd prefer to talk over Discord about it or continue in the issue thread.

`_itemLotParam_map` or `_itemLotParam_enemy` seems to not have data for a few weapons from the DLC. (While having data for the majority of DLC item lots).

EDIT: disregard this, I've found the root cause.

**Comment by ShaneD20** [comment]
An update, a few other weapons that seem to have the same issue.
- 15120000 Butchering Knife, Anastasia
- 9010000   Nagakiba, Yura
- 21070000 Iron Balls, Boggart
- 18030008 Edgar's Halberd
